app/src/main/java/it/niedermann/nextcloud/deck/persistence/sync/adapters/db/DataBaseAdapter.java

```diff
@@ -865,4 +866,24 @@ public void deleteSingleCardWidget(int widgetId) {
         model.setWidgetId(widgetId);
         db.getSingleCardWidgetModelDao().delete(model);
     }
+
+    public long createStackWidget(int appWidgetId, long accountId, long stackId, boolean darkTheme) {
+        StackWidgetModel model = new StackWidgetModel();
+        model.setAppWidgetId(appWidgetId);
+        model.setAccountId(accountId);
+        model.setStackId(stackId);
+        model.setDarkTheme(darkTheme);
+
+        return db.getStackWidgetModelDao().insert(model);
+    }
+
+    public StackWidgetModel getStackWidgetModel(int appWidgetId) {
+        return db.getStackWidgetModelDao().getStackWidgetByRemoteId(appWidgetId);
```

Comment:
The term `remoteId` is reserved for remote entities (from server). This would be a `localId`. Since there is no remote appWidget, i' suggest either `getStackWidgetByLocalId` or simply `getStackWidgetById`

Reply:
Or `...ByAppWidgetId`

Reply:
and `...Directly` :grinning: